src/js/components/Calendar.js

```diff
@@ -285,24 +301,43 @@ export default class Calendar extends Component {
   }
 
   render () {
-    var classes = [CLASS_ROOT];
-    if (this.state.dropActive) {
-      classes.push(CLASS_ROOT + "--active");
-    }
-    if (this.props.className) {
-      classes.push(this.props.className);
-    }
+    const {
+      className,
+      id,
+      name,
+      value,
+      onChange, // eslint-disable-line no-unused-vars
+      ...props
+    } = this.props;
+
+    const classes = classnames(
+      CLASS_ROOT,
+      className,
+      {
+        [`${CLASS_ROOT}--active`]: this.state.dropActive
+      }
+    );
 
     return (
-      <div ref="component" className={classes.join(' ')}>
-        <input className={CLASS_ROOT + "__input"}
-          id={this.props.id} ref="calendarInput" name={this.props.name}
-          value={this.props.value}
-          onChange={this._onInputChange} />
-        <Button className={CLASS_ROOT + "__control"} icon={<CalendarIcon
-          a11yTitle='calendar-icon-title'
-          a11yTitleId='calendar-icon-title-id' />}
-          onClick={this._onOpen} />
+      <div ref="component" className={classes} {...props}>
```

Comment:
Transferring props at the div level will propagate any event from children. is that the expected behavior?

Initially I was under the impression that transferring props would be placed in the input field only.

The behavior I've experienced when I've tested this PR was:

0 - Added onFocus to the calendar.
1 - Pressed tab until the Calendar search input
2 - Focus callback successfully invoked
3 - Pressed tab again, focus is now in the button
4 - Focus callback invoked again (because of the even propagation)

I think that transferring props in this component is very tricky as I'm not sure where is the right place to put it.

@ericsoderberghp your insight here would be great.